countComponents.js

```diff
@@ -0,0 +1,75 @@
+/*
+graph of nodes
+
+array[i] = [a, b]
+
+i: n of nodes
+o: => number of connected components
+
+Input: n = 5, edges = [[0,1],[1,2],[3,4]]
+Output: 2
+
+0 - 1 - 2
+
+3 - 4
+
+*/
+
+
+const buildGraph = (edges) => {
+    const graph = {};
+
+    for (const edge of edges) {
+        const [nodeOne, nodeTwo] = edge;
+
+        const isNodeOne = graph.hasOwnProperty(nodeOne);
```

Comment:
This name doesn't make sense. I'd expect this variable to tell me if something "is node one". It doesn't do that. It tells me if the graph contains a property named `nodeOne`. Instead, it would be better to name this `hasNodeOne` or `containsNodeOne`